R/predict.sgdnet.R

```diff
@@ -222,28 +225,34 @@ predict.sgdnet <- function(object,
     dimnames(beta) = list(vnames, paste(seq(along = s)))
   }
 
+  if (type %in% c("link", "response", "class")) {
+    if (inherits(newx, "sparseMatrix"))
+      newx <- methods::as(newx, "dgCMatrix")
+    else
+      newx <- as.matrix(newx)
+
+    linear_prediction <- as.matrix(cbind(1, newx) %*% beta)
```

Comment:
Typically called the "linear predictor" not prediction since it may not be the right "sort of thing" to be a prediction (e.g., it can be negative even if y is 0/1), but not a big deal.